### PR TITLE
 hpc instructions to use mamba and small fixes

### LINKED ISSUE #825: Modify HPC instructions to use base env for jupyterlab
I would like to modify the [Pangeo software environment instructions for HPC](https://pangeo.io/setup_guides/hpc.html#installing-a-software-environment) to be more aligned with what I recommend people do, but wanted to check to see if others agree.  

The basic procedure is:
* install miniconda  (unchanged)
* specify conda-forge as the only channel in the config
* install mamba (for faster dependency solving)
* update all packages in the base env
* in the base env, install the jupyterlab stuff (jupyterlab, jupyter-server-proxy, dask-labextension, ipywidgets, nb_conda_kernels, nodejs)
* create one or more custom environments that contain other packages like (dask, dask-jobqueue, xarray, holoviz, ipykernel, ipywidgets and extra packages). 
* pick a random port to use on the server instead of assuming 8888 will be available. 
* launch jupyterlab from the base environment on that port

Would it be okay for me to submit a PR with these changes, or is this too opinionated (or someone has a better best practice)?

### PR DESCRIPTION
**Pull request by rsignell-usgs**
fixes #825

### CODE REVIEW

File: docs/setup_guides/hpc.rst [hunk 1]
```diff
@@ -34,12 +34,19 @@ After you have logged into your HPC system, download and install Miniconda:
...
+
+Before creating your environment, update your conda package manager with packages 
+from the conda-forge channel instead of the default channel and install Mamba, 
+which works like conda but is written in C++ and therefore creates environments faster. 
 
 ::
     
-    conda update conda
+    conda config --add channels conda-forge --force
+    conda config --remove channels defaults --force
```
Comment (guillaumeeb) [inline docs/setup_guides/hpc.rst:46]: I'm not sure why you need to use only conda-forge? This is to ensure the best compatibility between all the dependencies. This config might prevent using other channels like pytorch, don't this? 
But at this point if a user needs other channels he knows what he's doing, so maybe that's OK.

File: docs/setup_guides/hpc.rst [hunk 2]
```diff
@@ -175,37 +181,47 @@ From here, we can start jupyter. The Cheyenne computer administrators have
...
+    JPORT=$(shuf -i 8400-9400 -n 1)
+    echo ""
+    echo ""
+    echo "Step 1: Wait until this script says the Jupyter server"
+    echo "        has started. "
+    echo ""
+    echo "Step 2: Copy this ssh command into a terminal on your"
+    echo "        local computer:"
+    echo ""
+    echo "        ssh -N -L 8889:`hostname`:$JPORT $USER@poseidon.whoi.edu"
```
Comment (guillaumeeb) [inline docs/setup_guides/hpc.rst:201]: `poseidon.whoi.edu`: maybe replace with a generic name.

**Review by guillaumeeb — COMMENTED** [review]
Just two remarks, otherwise this looks really good!

File: docs/setup_guides/hpc.rst [hunk 3]
```diff
@@ -175,37 +181,47 @@ From here, we can start jupyter. The Cheyenne computer administrators have
...
+    [I 2021-04-06 06:33:57.962 ServerApp] Jupyter Server 1.5.1 is running at:
+    [I 2021-04-06 06:33:57.962 ServerApp] http://pn009:8537/lab     
+    [I 2021-04-06 06:33:57.963 ServerApp]  or http://127.0.0.1:8537/lab
+    [I 2021-04-06 06:33:57.963 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
 
-You'll want to change the details in the command above but the basic idea is
-that we're passing the port 8888 from the compute node `r8i4n0` to our
-local system. Now open http://localhost:8888 on your local machine, you should
-find a jupyter server running!
+Just follow the Steps 1,2,3 printed out by the script to get connected.
```
Comment (andersy005) [inline docs/setup_guides/hpc.rst:224]: 📣 Shameless plug 📣 ;) : You could replace all of this with a **single command** via [jupyter-forward](https://github.com/NCAR/jupyter-forward) - Jupyter-forward launches Jupyter Lab over SSH (on any remote machine) with a single command !

Here are some examples:

- Launching Jupyter Lab on a remote host’s login node: https://jupyter-forward.readthedocs.io/en/latest/examples.html#launching-jupyter-lab-on-a-remote-host-s-login-node
- Launching Jupyter Lab on a remote host’s compute node: https://jupyter-forward.readthedocs.io/en/latest/examples.html#launching-jupyter-lab-on-a-remote-host-s-compute-node

Comment (rsignell-usgs) [reply docs/setup_guides/hpc.rst:224]: @andersy005 , what do you think about leaving these steps in here, and then saying that many of them can be automated by jupyter-forward as long as you are not on windows?   

I'm hesitant to just replacing this with jupyter-forward recommendation until https://github.com/NCAR/jupyter-forward/issues/81 is figured out.

File: docs/setup_guides/hpc.rst (hunk 1 above)
Comment (ocefpaf) [reply docs/setup_guides/hpc.rst:46]: Any kind of channel mixing is problematic. We always recommend to use only 1 and, if it is extremely necessary, add the extra channel in the env file configuration. However, the ideal of keeping defaults and conda-forge is usually problematic, specially without strict channel policy. That is b/c those two channels overlap a lot. It is not like getting pytorch from another channel, it is like mixing from two different sources for all packages.

The other advantage, besides avoiding compatibility issues, is to speed up the solver time by reducing the amount of packages it has to work with.

Comment (rsignell-usgs) [reply docs/setup_guides/hpc.rst:46]: Thanks for the clarification Filipe!  This has been a moving target for many of us, so good to know that one channel is the way to go now!

File: docs/setup_guides/hpc.rst (hunk 3 above)
Comment (rsignell-usgs) [reply docs/setup_guides/hpc.rst:224]: @andersy005 , what do you think about just merging this and following it with something about jupyter forward as another PR?

**Comment by guillaumeeb** [comment]
I'll just go ahead and merge this one!

**Pull request merged**